ibmdbpy/geoFrame.py

```diff
@@ -1012,29 +1014,43 @@ def _binary_operation_handler(self, ida2, db2gse_function,
         -------
         IdaGeoDataFrame
         """
-        ida1 = self  # For code clearness
+        ida1 = self
+        
+        # Check if allowed data type
         if not (ida1.dtypes.TYPENAME[0] in valid_types_ida1 or
                         valid_types_ida1[0] == 'ST_GEOMETRY'):
             raise TypeError("Column " + ida1.column +
                             " has incompatible type.")
         if not (ida2.dtypes.TYPENAME[0] in valid_types_ida2 or
                         valid_types_ida2[0] == 'ST_GEOMETRY'):
-            raise TypeError("Column " + ida1.column +
+            raise TypeError("Column " + ida2.column +
                             " has incompatible type.")
 
         # Get the definitions of the columns, which will be the arguments for
         # the DB2GSE function
-        column1_for_db2gse = ida1.internal_state.columndict[self.geometry.column]
+        column1_for_db2gse = ida1.internal_state.columndict[ida1.geometry.column] 
         if column1_for_db2gse[0] == '\"' and column1_for_db2gse[-1] == '\"':
             column1_for_db2gse = column1_for_db2gse[1:-1]
-        column2_for_db2gse = ida2.internal_state.columndict[self.geometry.column]
+        column2_for_db2gse = ida2.internal_state.columndict[ida2.geometry.column] 
         if column2_for_db2gse[0] == '\"' and column2_for_db2gse[-1] == '\"':
             column2_for_db2gse = column2_for_db2gse[1:-1]
-
+        
+        ### added : check if eligible column names ###
+        # in order to fix error occuring when geometry column defined as function(other_column)
+        # geometry columns must have been explicitly defined and have an eligible alias
+        print("column1_for_db2gse: "+column1_for_db2gse)
+        print("column2_for_db2gse: "+column2_for_db2gse)
+        if "(" in column1_for_db2gse or "ST_" in column1_for_db2gse:
+            print("A function pattern was detected in column1_for_db2gse. Please provide the alias of the first geometry column.")
+            column1_for_db2gse = str(input("ALIAS_1: "))
```

Comment:
Like the previous comment, please don't use `input`. Just raise an exception, like:

```python
raise ValueError("A function pattern was detected in column1_for_db2gse ('" + column1_for_db2gse + "').")
```

Reply:
![image](https://user-images.githubusercontent.com/55744208/74664721-8964f900-519e-11ea-934c-9e917855fcf2.png)
I understand it is not practical to use `input`here. However how do you suggest tackling this issue otherwise? Let me explain:

In the example above, query should be:
CREATE VIEW "TEST_VIEW" AS 
(SELECT IDA1."USER_ID" AS "INDEXERIDA1",IDA2."ID" AS "INDEXERIDA2",DB2GSE.ST_CONTAINS(DB2GSE.ST_BUFFER(IDA1.USER_LOC,50),IDA2.GEOM) AS "RESULT" 
FROM 
(SELECT "USER_ID","USER_LOC","MUSEUM_TYPE","MAX_DIST",DB2GSE.ST_BUFFER(USER_LOC,50) AS "DEFAULT_BUFFER" FROM BLUADMIN.GEO_TOURISTS) AS IDA1, 
BLUADMIN.GEO_MUSEUMS AS IDA2);
But due to the structure of the code, you cannot change “IDA1.DB2GSE_FUNCTION(COL, ARG)” into “DB2GSE_FUNCTION(IDA1.COL,ARG)” automatically because it is derived from other queries. So the fix relied on the user input: if an intricated function pattern is detected in the query, user input is asked to avoid confusion. An explanation is prompted and the user is asked for the geometry column name. 
Do you see another way to resolve this issue?

Reply:
in this specific case, raising an error prevents the user to apply several geospatial methods in a row on the same data

Reply:
I have modified the code to avoid user input, I have seen it with Toni, it works :)